### DISCUSSION TITLE
Error when creating a custom verifiable credential (Context parameter)

### DISCUSSION THREAD

**Discussion by soufianeAmaador** [walt-id/.github#34] [Help] [answered]
Hi all, 
I am trying to create a custom verifiable credentials with the VC.Creator website provided by Walt.id, and when trying to call the function `encodedCustomVC.toCredential()`, it throws an error regarding the variable context.

The error states:

> Exception in thread "main" com.beust.klaxon.KlaxonException: Unable to instantiate VerifiableAttestation:
    Parameter context: expected kotlin.collections.List<kotlin.String> but received java.util.ArrayList (value: [https://www.w3.org/2018/credentials/v1])

I find this very weird because the **context** **variable**  in the data class of the custom verifiable credential I implemented is of type `kotlin.collections.List`. Somehow when calling the function it throws an error saying the context variable is of type `java.util.ArrayList`.
I tried explicitly typing the Kotlin.collections.List: `    @Json(name = "@context") var context: kotlin.collections.List<kotlin.String> = listOf("https://www.w3.org/2018/credentials/v1"),
` But it still throws the same error.

![CONTEXT-TYPE-CONVERSION](https://user-images.githubusercontent.com/70653226/212494542-afca6afd-6bf5-451f-ae46-100ad7028310.png)
![CONTEXT-TYPE-CONVERSION-2](https://user-images.githubusercontent.com/70653226/212494545-39a6f964-4588-4a70-ae63-5aea56bc099e.png)

**Comment by severinstampler** [comment, marked as answer]
Hi @soufianeAmaador 
As it seems the parser tries to deserialize a VerifiableAttestation, not your custom credential. Make sure the credential contains the correct type property, such that the parser can find the correct class for deserialization.

Having said that, I wonder if you're using the vclib directly in your project, or whether you use the ssikit.
Because, the ssikit has recently been refactored with regards the credential serialization, deserialization and template management. It no longer depends on vclib at all and allows you to import any credential as template dynamically from a JSON sample of that credential.
You may want to look into this instead.

Hope I could help, best regards
Severin
